### ISSUE TITLE
Uncaught TypeError: Cannot set property '0' of undefined

### ISSUE THREAD

**Issue by clown-helang** [apache/echarts#13441]
### Version
echarts@4.9.0

### Reproduction link
[https://codepen.io/HeLang582/pen/QWENEom?editors=1011](https://codepen.io/HeLang582/pen/QWENEom?editors=1011)

### Steps to reproduce
1. Create a line graph to create horizontal scrolling effect when too much data is set.(创建一个折线图，设置数据过多时出现水平滚动效果。)
2. Drag the horizontal scroll bar back and forth repeatedly to reproduce the error, the X-axis coordinates will overlap, and the rendering of the value will be chaotic(反复来回拖动水平滚动条即可复现报错，x轴坐标会出现重叠现象，值的渲染也会出现混乱)

### What is expected?
Graphics rendering correctly, no error reporting(图形渲染正确，不要有报错)

### What is actually happening?
Through the error checking in the Console, it was found that there is an interpolateArray method in the Animator file of ZRender, the length of the parameter Out passed at the time of invocation is not the same as the length of P0 and P1, resulting in an error in the assignment, so we need to do an interpolating processing: the following code(通过console中报错检查，发现在ZRender的Animator文件中有一个interpolateArray方法，在调用执行的时候传给的参数out的长度和p0以及p1的长度不一样,导致赋值报错, 所以需要做一下兼容处理：具体代码如下：):
```
function interpolateArray(p0, p1, percent, out, arrDim) {
  
  var len = p0.length;

  if (arrDim === 1) {
    for (var i = 0; i < len; i  ) {
      out[i] = interpolateNumber(p0[i], p1[i], percent);
    }
  } else {
    var len2 = len && p0[0].length;

    for (var i = 0; i < len; i  ) {
      for (var j = 0; j < len2; j  ) {
		/* 
		* 由于out的length可能和p0, p1的length不一样导致out[i][j]赋值的时候会报错，
		* 所以需要添加一个if判断，如果out[i]不存在，则给out[i]赋值一个空数组，这样可以保证代码不报错
		*/
		if (!out[i]) out[i] = [];
        out[i][j] = interpolateNumber(p0[i][j], p1[i][j], percent);
      }
    }
  }
}
```

Through the above compatibility processing, normal data update of ECharts can be guaranteed without the problem of horizontal coordinate overlap. Fundamentally speaking, there should be some bugs in echarts data rendering. Since it is difficult to check, I did not locate the root cause.So I hope the authorities can fix this problem（通过上面的兼容处理可以保证echarts正常的数据更新不会出现横坐标重叠的问题，从根本上讲应该是echarts数据渲染上存在有BUG，由于不好排查的，所以我没有去定位根因。所以希望官方可以修复一下这个问题。）.

---
The problem can be reproduced on codePen, which is a relatively serious problem, resulting in overlapping problems in chart rendering（问题可以在codePen上复现，是一个较为严重的问题，会导致图表渲染出现重叠问题）.

**Comment by clown-helang** [comment]
![1](https://user-images.githubusercontent.com/20653395/96250289-33707b80-0fe1-11eb-87ca-7559d4f86796.gif)

**Comment by plainheart** [comment]
Seems to be related to #12815, and @pissang pushed a commit https://github.com/ecomfe/zrender/commit/94baa6ee141ce3d281ba1e02b495834adb49a25f that should have fixed this issue.

**Comment by clown-helang** [comment]
Will this release 4.9.x minor fix this problem?

**Comment by plainheart** [comment]
I'm afraid no versions will be released anymore in the 4.x branch. The next version will be 5.0 which should be released this month as planned.

**Comment by clown-helang** [comment]
May I ask when this BUG can be fixed

**Comment by clown-helang** [comment]
It looks like it will not report an error.But there is a new problem, if you drag faster, there will be a rendering data update problem。

https://codepen.io/clown-helang/pen/QWENEom?editors=1011

**Comment by clown-helang** [comment]
![image](https://user-images.githubusercontent.com/20653395/105135603-4a1bf680-5b2b-11eb-9037-43e7bd7e18d0.png)

**Comment by Archdemonllllll** [comment]
I also had this problem， is it solved?

**Comment by clown-helang** [comment]
Upgrade to version 5.x has not been replicated

**Comment by shizhihuaxu** [comment]
Hopefully this will be fixed in 4.x

**Comment by Ovilia** [comment]
Seem to be fixed in 5.0.1
